src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenCallTests.vb

```diff
@@ -7727,6 +7727,8808 @@ Position set for item '-4'
+  IL_0024:  castclass  "Integer()"
+  IL_0029:  constrained. "T"
+  IL_002f:  callvirt   "Sub IMoveable.set_Position(Integer, Integer())"
+  IL_0034:  ret
+}
+]]>)
+
+            'Wrong IL
+            verifier.VerifyIL("Program.Call2(Of T)",
+            <![CDATA[
+{
+  // Code size       53 (0x35)
+  .maxstack  4
+  .locals init (Integer V_0)
+  IL_0000:  ldarga.s   V_0
+  IL_0002:  ldarga.s   V_0
+  IL_0004:  call       "Function Program.GetOffset(Of T)(ByRef T) As Integer"
+  IL_0009:  dup
+  IL_000a:  stloc.0
+  IL_000b:  ldarga.s   V_0
+  IL_000d:  ldloc.0
+  IL_000e:  constrained. "T"
+  IL_0014:  callvirt   "Function IMoveable.get_Position(Integer) As Integer()"
+  IL_0019:  ldc.i4.2
+  IL_001a:  newarr     "Integer"
+  IL_001f:  call       "Function Microsoft.VisualBasic.CompilerServices.Utils.CopyArray(System.Array, System.Array) As System.Array"
+  IL_0024:  castclass  "Integer()"
+  IL_0029:  constrained. "T"
+  IL_002f:  callvirt   "Sub IMoveable.set_Position(Integer, Integer())"
+  IL_0034:  ret
+}
+]]>)
+        End Sub
+
+        <Fact>
+        <WorkItem(63221, "https://github.com/dotnet/roslyn/issues/63221")>
+        Public Sub GenericTypeParameterAsReceiver_Redim_Indexer_Struct_Index()
+            Dim comp =
+<compilation>
+    <file>
+Imports System
+
+Interface IMoveable
+    Property Position(x As Integer) As Integer()
+End Interface
+
+Structure Item
+    Implements IMoveable
+
+    Public Property Name As String
+
+    Public Property Position(x As Integer) As Integer() Implements IMoveable.Position
+        Get
+            Console.WriteLine("Position get for item '{0}'", Me.Name)
+            Return Nothing
+        End Get
+        Set
+            Console.WriteLine("Position set for item '{0}'", Me.Name)
+        End Set
+    End Property
+End Structure
+
+Class Program
+    Shared Sub Main()
+        Dim item1 = New Item With {.Name = "1"}
+        Call1(item1)
+
+        Dim item2 = New Item With {.Name = "2"}
+        Call2(item2)
+    End Sub
+
+    Private Shared Sub Call1(Of T As {Structure, IMoveable})(item As T)
+        Redim Preserve item.Position(GetOffset(item))(1)
+    End Sub
+
+    Private Shared Sub Call2(Of T As {IMoveable})(item As T)
+        Redim Preserve item.Position(GetOffset(item))(1)
+    End Sub
+
+    Shared value as Integer
+
+    Shared Function GetOffset(Of T)(ByRef item As T) As Integer
+        value -= 1
+        item = DirectCast(DirectCast(New Item With {.Name = value.ToString()}, IMoveable), T)
+        Return Nothing
+    End Function
+End Class
+    </file>
+</compilation>
+
+            Dim verifier = CompileAndVerifyEx(comp, targetFramework:=TargetFramework.StandardAndVBRuntime, options:=TestOptions.ReleaseExe, expectedOutput:=
+"
+Position get for item '-1'
+Position set for item '-1'
+Position get for item '-2'
+Position set for item '-2'
+").VerifyDiagnostics()
+
+            verifier.VerifyIL("Program.Call1(Of T)",
+            <![CDATA[
```

Comment:
> Not verifying Call2? Is that because it's a boxed value so there's nothing new from the last test?

The method is exactly the same as in the previous test and its IL is verified there. We do verify expected behavior when the method is invoked for a structure.